bravado/client.py

```diff
@@ -350,78 +193,21 @@ def from_url(
         """
         log.debug(u"Loading from %s" % url)
         http_client = http_client or SynchronousHttpClient()
-
-        # TODO: better way to customize the request for api-docs, so we don't
-        # have to add new kwargs for everything
         loader = Loader(http_client, request_headers=request_headers)
-
-        # Loads and validates the spec
-        spec = loader.load_spec(url)
-
-        return cls.from_spec(spec, http_client, origin_url=url)
-
-    # @classmethod
-    # def from_resource_listing(
-    #         cls,
-    #         resource_listing,
-    #         http_client=None,
-    #         api_base_path=None,
-    #         url=None):
-    #     """
-    #     Build a :class:`SwaggerClient` from swagger api docs
-    #
-    #     :param resource_listing: a dict with a list of api definitions
-    #     :param http_client: an HTTP client used to perform requests
-    #     :type  http_client: :class:`bravado.http_client.HttpClient`
-    #     :param api_base_path: a url, override the path used to make api
-    #       requests
-    #     :type  api_base_path: str
-    #     :param api_doc_request_headers: Headers to pass with api docs requests
-    #     :type  api_doc_request_headers: dict
-    #     :param url: the url used to retrieve the resource listing
-    #     :type  url: str
-    #     """
-    #     url = url or resource_listing.get(u'url')
-    #     log.debug(u"Using resources from %s" % url)
-    #
-    #     if url:
-    #         url_base = "{uri.scheme}://{uri.netloc}".format(uri=urlparse(url))
-    #     else:
-    #         url_base = None
-    #
-    #     resources = build_resources_from_spec(
-    #         http_client or SynchronousHttpClient(),
-    #         map(append_name_to_api, resource_listing['apis']),
-    #         api_base_path,
-    #         url_base)
-    #     return cls(url, resources)
+        spec_dict = loader.load_spec(url)
+        return cls.from_spec(spec_dict, origin_url=url, http_client=http_client)
 
     @classmethod
-    def from_spec(cls, spec, http_client=None, origin_url=None):
+    def from_spec(cls, spec_dict, origin_url=None, http_client=None):
         """
         Build a :class:`SwaggerClient` from swagger api docs
 
-        :param spec: a dict with a Swagger spec in json-like form
-        :param http_client: an HTTP client used to perform requests
-        :type  http_client: :class:`bravado.http_client.HttpClient`
```

Comment:
The `http_client` is now an attribute of the `bravado.mapping.spec.Spec` class so it is not retained by `SwaggerClient` directly.